src/timesigglyph.ts

```diff
@@ -0,0 +1,85 @@
+import { Glyph, GlyphMetrics } from './glyph';
+import { TimeSignature } from './timesignature';
+
+export class TimeSignatureGlyph extends Glyph {
+  timeSignature: TimeSignature;
+  topStartX: number;
+  botStartX: number;
+  width: number;
+  xMin: number;
+
+  constructor(
+    timeSignature: TimeSignature,
+    topDigits: string[],
+    botDigits: string[],
+    code: string,
+    point: number,
+    options?: { category: string }
+  ) {
+    super(code, point, options);
+    this.timeSignature = timeSignature;
+    this.topGlyphs = [];
+    this.botGlyphs = [];
+
+    let topWidth = 0;
+    for (let i = 0; i < topDigits.length; ++i) {
+      const num = topDigits[i];
+      const topGlyph = new Glyph('timeSig' + num, this.timeSignature.point!);
+
+      this.topGlyphs.push(topGlyph);
+      topWidth += topGlyph.getMetrics().width!;
+    }
+
+    let botWidth = 0;
+    for (let i = 0; i < botDigits.length; ++i) {
+      const num = botDigits[i];
+      const botGlyph = new Glyph('timeSig' + num, this.timeSignature.point!);
+
+      this.botGlyphs.push(botGlyph);
+      botWidth += botGlyph.getMetrics().width!;
+    }
+
+    this.width = topWidth > botWidth ? topWidth : botWidth;
+    this.xMin = this.getMetrics().x_min;
```

Comment:
this just returns the current value of xMin, so it's the same as `this.xMin = this.xMin`.

Reply:
`this.xMin` removed, now using `this.getMetrics().x_min`

Reply:
I had to undo this change and leave as it was. Somehow `this.getMetrics().x_min` changes its value in excution time.